app/javascript/services/courses-service.js

```diff
@@ -0,0 +1,10 @@
+import axios from '../helpers/axios-rails';
+import {courseJoinRoute} from "./service-routes";
+
+class CoursesService {
+    static async joinCourse(courseId) {
+        return axios.post(courseJoinRoute(courseId)).then(response=>response.data);
+    }
+}
+
+export default CoursesService;
```

Comment:
I'm not sure I understand how this code relates to the rest... but maybe it will become clear as I read on.

Reply:
After checking with Brian, he confirmed that this should probably be in a different PR... Ahsun also suggested refactoring to use `await` instead of `then`.    Faiz notes that this is an existing pattern in the code that should probably be refactored.  Also probably doesn't need to be `static`